android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetViewModel.kt

```diff
@@ -16,107 +16,57 @@
-  }
-
-  fun getFamiliesFeatureCollectionStream(context: Context): LiveData<FeatureCollection> {
-    val featureCollectionLiveData = MutableLiveData<FeatureCollection>()
-
-    viewModelScope.launch(dispatcherProvider.io()) {
-      val familyFeatures = getFamiliesFeatureCollection(context)
-      featureCollectionLiveData.postValue(familyFeatures)
-    }
-
-    return featureCollectionLiveData
-  }
-
-  suspend fun getFamilies(): List<Pair<Group, Location>> {
-    val coding =
-      Coding().apply {
-        system = "http://hl7.org/fhir/group-type"
-        code = "person"
+  fun addLocationToMap(location: GeoWidgetLocation) {
+    val contexts = location.contexts.map { context ->
+      JSONObject().apply {
+        put("id", context.id)
+        put("type", context.type)
       }
+    }
+    val properties = JSONObject().apply {
+      put("id", location.id)
+      put("name", location.name)
+      put("contexts", JSONArray(contexts))
+    }
 
-    val familiesWithLocations =
-      defaultRepository.fhirEngine
-        .search<Group> { filter(Group.TYPE, { value = of(coding) }) }
-        .asSequence()
-        .map { it.resource }
-        .filter {
-          // it.hasExtension("http://build.fhir.org/extension-location-boundary-geojson.html")
-          it.characteristic.firstOrNull { characteristic ->
-            characteristic.value is Reference &&
-              characteristic.valueReference.reference.contains(ResourceType.Location.name)
-          } != null
-        }
-
-    val familiesList = ArrayList<Pair<Group, Location>>()
-
-    familiesWithLocations.forEach { family ->
-      try {
-        val familyLocation = defaultRepository.fhirEngine.get<Location>(familyLocationId(family))
-
-        familiesList.add(Pair(family, familyLocation))
-      } catch (ex: ResourceNotFoundException) {
-        Timber.e(ex)
+    val jsonFeature =
+      JSONObject().apply {
+        put("type", "Feature")
+        put("properties", properties)
+        put("geometry", location.getGeoJsonGeometry())
       }
-    }
+    val feature = Feature.fromJson(jsonFeature.toString())
 
-    return familiesList
+    _featuresFlow.value = _featuresFlow.value + feature
   }
 
-  private fun familyLocationId(family: Group) =
-    family.characteristic
-      .firstOrNull { characteristic ->
-        characteristic.value is Reference &&
-          characteristic.valueReference.reference.contains("Location")
-      }!!
-      .valueReference
-      .referenceElement
-      .idPart
-
-  fun saveLocation(location: Location): LiveData<Boolean> {
-    val liveData = MutableLiveData<Boolean>()
-    viewModelScope.launch(dispatcherProvider.io()) {
-      defaultRepository.create(true, location)
-      liveData.postValue(true)
+  fun addLocationsToMap(locations: Set<GeoWidgetLocation>) {
+    locations.forEach { location ->
+      addLocationToMap(location)
     }
+  }
 
-    return liveData
+  fun clearLocations() {
+    _featuresFlow.value = setOf()
```

Comment:
Is this one-liner function used in more than one place? If only used in one place you can move the single line of code to where the function is invoked.

Reply:
this will be called from multiple places.